File: Results/DER/Translation/translation_output/no_test/starcoder/codenet/python/java/p03211.java
<fim_prefix>Translate the following python code to java and enclose your solution inside ```java```:
```
S=str(input())
a=99999
for i in range(len(S)-2):
  a=min(abs(753-int(S[i]+S[i+1]+S[i+2])),a)
  
print(a)
```

java code:<fim_suffix><fim_middle>
```java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        String s=sc.next();
        int a=99999;
        for(int i=0;i<s.length()-2;i++){
            a=Math.min(Math.abs(753-Integer.parseInt(s.substring(i,i+3))),a);
        }
        System.out.println(a);
    }
}
```
<|endoftext|>

File: Results/DER/Translation/translation_output/no_test/deepseekcoder/codenet/java/python/p03211.py
<｜begin▁of▁sentence｜>You are an expert Python programmer.Your task is to write a Python function to solve a programming problem.

### Instruction:
Translate the following java code to python and enclose your solution inside ```python```:
```
import java.util.Scanner;

public class p03211 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String S = sc.next();
        int min = 1000;
        for (int i = 0; i < S.length() - 2; i++) {
            int X = Integer.parseInt(S.substring(i, i + 3));
            min = Math.min(min, Math.abs(X - 753));
        }
        System.out.println(min);
        sc.close();
    }
}
```

### Response:
```python
import sys

def main():
    S = input()
    min_diff = sys.maxsize
    for i in range(len(S) - 2):
        X = int(S[i:i + 3])
        min_diff = min(min_diff, abs(X - 753))
    print(min_diff)

if __name__ == "__main__":
    main()
```
<|EOT|>
